### ISSUE TITLE
Allow exercises to build and run on .NET Core

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/xcsharp#200]
In [this PR](https://github.com/exercism/xcsharp/pull/199), you can find the work to get our exercises running on .NET Core. This issue will keep track of the work items.

- [x] Convert tests from NUnit to xUnit (#193)
- [x] Create a project for each exercise (#198)
- [x] Fix the CI builds
- [x] Create default file for each exercise (#138)
- [ ] Update the documentation (#146)

Once these items have been completed, we probably want to to wait at least until the final release of Visual Studio 2017 (which will be on march 7th) and the RTM version of the .NET CLI tooling. In the mean time, I would really really appreciate it if people would be willing to look at this PR and hopefully also test things locally.

To make testing and editing all projects at the same time easier, I've also added a solution file that contains all projects.

When this PR has finally been merged, we can then move on to issue #138, where we add a default file for each exercise, possibly also with the required class/methods also having an empty, stub implementation.

**Comment by ErikSchierboom** [comment]
I'm slowly getting there :) I now have a separate project for each exercise that runs on .NET Core. The CI servers now also succesfully run the .NET Core tests.

**Comment by ErikSchierboom** [comment]
Things are looking good! I've now managed to get the latest build working on .NET Core on all three CI servers. This build includes the following features:

- Project file per exercise
- Default (stub) file per exercise
- Project file is in the new .csproj format
- Project can be built using .NET Core
- Multi-platform support due to .NET Core compatibility
- Migrated tests from NUnit to xUnit

The main thing that is still lacking is the documentation, but for that I want to wait until both the .NET CLI and VS 2017 are RTM. Once they have been released, we should update the documentation and finish the migration :)
